### DISCUSSION TITLE
Issue: H3HexagonLayer requires h3-js, but h3-js import via npm fails in Observable

### DISCUSSION THREAD

**Discussion by achirkunov** [observablehq/framework#1774] [Q&A] [answered]
[As per documentation](https://deck.gl/docs/api-reference/geo-layers/h3-hexagon-layer), H3HexagonLayer requires `h3-js` to be included before `deck.gl`. However, importing h3-js using Observable's npm import mechanism results in the h3-js functions not being available to deck.gl.

### Reproduction steps
When trying to import `h3-js` via npm like this:

```js
import * as h3 from "npm:h3-js";
import deck from "npm:deck.gl";
const {DeckGL, H3HexagonLayer, HexagonLayer} = deck;
```

Observable loads the following resource for h3-js (version 4.1.0):
```html
<link rel="modulepreload" href="./_npm/h3-js@4.1.0/_esm.js">
```

But when the H3HexagonLayer is instantiated, deck.gl throws an error as it attempts to call h3-js functions:
```js
deck: initialization of H3HexagonLayer({id: 'heatmap'}): (0,EC.getResolution) is not a function. (In '(0,EC.getResolution)(r)', '(0,EC.getResolution)' is undefined)
deck: initialization of ColumnLayer({id: 'heatmap-hexagon-cell'}): (0,wC.cellToLatLng) is not a function. (In '(0,wC.cellToLatLng)(r)', '(0,wC.cellToLatLng)' is undefined)
```
I have checked that h3 is available using `console.log(h3)`:
```js
Module {UNITS: Object, areNeighborCells: function, cellArea: function, cellToBoundary: function, cellToCenterChild: function, …}
```

### Temporary workaround
Importing h3-js using a <script> tag works as expected, and H3HexagonLayer can successfully access the h3-js functions. For example:

````markdown
<script src="https://unpkg.com/h3-js@^4.1.0"></script>
```js
import deck from "npm:deck.gl";
const {DeckGL, H3HexagonLayer, HexagonLayer} = deck;
```
````

### Concern
While using the `<script>` tag resolves the issue, Observable recommends importing dependencies via the import statement.

**Comment by Fil** [comment, marked as answer]
This is an issue with the way Deck expects h3 to be a global (window) property, not a Framework bug.

A workaround:

~~~~
```js
import * as h3 from "npm:h3-js";
window.h3 = h3;
```

```js
h3;
import deck from "npm:deck.gl";
const {Deck, DeckGL, H3HexagonLayer, HexagonLayer} = deck;
```
~~~~

Note: we need two separate fenced blocks, because [imports declarations are hoisted](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and we have to ensure that the (questionable) `window.h3 = h3;` runs before the deck import.

**Reply by achirkunov** [reply]
Thank you Fil - works perfectly now.

**Reply by Fil** [reply]
Thanks. I've made a more complete example (including H3, DeckGL and maplibregl) here: https://observablehq.observablehq.cloud/pangea/party/h3-deck
